test/environment_unittest.cc

```diff
@@ -42,6 +49,59 @@ class TemporaryFile {
  private:
   boost::filesystem::path path_;
 };
+
+// Starts a server in a separate thread, allowing it to choose an
+// available port.
+class FakeServerThread {
+ public:
+  FakeServerThread()
+    : server_(Server::options(handler_).address("127.0.0.1").port("")) {
+    server_.listen();
```

Comment:
This is optional, right?

Reply:
Yeah, but it reserves a port, so the client can call GetUrl() right away.